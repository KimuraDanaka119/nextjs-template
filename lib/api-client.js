import fetch from 'isomorphic-unfetch';
import config from 'configs/config';

export default class ApiClient {
  async doFetch(path) {
    const url = config.api.baseURL + path;

    const res = await fetch(url);
    const data = await res.json();

    return {
      statusCode: res.status,
      data,
    };
  }

  async getShows(keyword) {
    const res = await this.doFetch(`/search/shows?q=${keyword}`);

    return res;
  }

  async getShow(id) {
    const res = await this.doFetch(`/shows/${id}`);

    return res;
  }
}
