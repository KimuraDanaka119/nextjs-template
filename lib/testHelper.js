import ApiClient from './api-client';

function t(key) {
  return key;
}

const config = {
  env: 'test',
  versionHash: 'test',
  paths: {
    static: '/static/test',
    styles: '/static/test/styles',
    images: '/static/test/images',
    locales: '/static/test/locales',
  },
  api: {
    baseURL: '.',
  },
};

const apiClient = new ApiClient('.');

export { t, config, apiClient };
