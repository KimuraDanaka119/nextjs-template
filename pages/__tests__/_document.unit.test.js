import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Document from '../_document';
import { config } from '../../lib/testHelper';

describe('<Document />', () => {
  it('should render correctly', () => {
    const document = shallow(<Document config={config}/>);
    expect(toJson(document)).toMatchSnapshot();
  });
});
