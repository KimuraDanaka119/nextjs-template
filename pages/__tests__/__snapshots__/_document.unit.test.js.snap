// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Document /> should render correctly 1`] = `
<html>
  <Head>
    <meta
      content="initial-scale=1.0, width=device-width"
      key="viewport"
      name="viewport"
    />
    <link
      href="/static/test/images/favicon.png"
      rel="shortcut icon"
    />
    <link
      crossOrigin="anonymous"
      href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/css/bootstrap.min.css"
      integrity="sha384-Zug+QiDoJOrZ5t4lssLdxGhVrurbmBWopoEl+M6BdEfwnCJZtKxi1KgxUyJq13dy"
      rel="stylesheet"
    />
    <link
      href="/static/test/styles/bundle.css"
      rel="stylesheet"
    />
  </Head>
  <body>
    <Main />
    <NextScript />
  </body>
</html>
`;
